src/indexApi.ts

```diff
@@ -1,18 +1,108 @@
 // BASE SETUP
 // =============================================================================
 
+import { asObject, asString } from 'cleaners'
 import cluster from 'cluster'
+import { formatISO } from 'date-fns'
 import { forkChildren, rebuildCouch } from 'edge-server-tools'
 
 import { config } from './utils/config'
-import { couchSchema } from './utils/couchSchema'
+import { couchSchema, minAmtDbPrefix } from './utils/couchSchema'
+import { mockPlugins } from './utils/mockPlugins'
 
 const express = require('express')
 const http = require('http')
+const nano = require('nano')
+const promisify = require('promisify-node')
+
+const asFulfilledPromise = asObject({
+  status: asString,
+  value: asObject({
+    _id: asString,
+    data: asObject({ minAmount: asString })
+  })
+})
 
 // call the packages we need
 const app = express()
 
+// Nano for CouchDB
+// =============================================================================
+const nanoDb = nano(config.dbFullpath)
+const swapMinAmts = mockPlugins.map(plugin => {
+  const pluginMinAmt = nanoDb.db.use(minAmtDbPrefix + plugin.pluginName)
+  return promisify(pluginMinAmt)
+})
+
+// ROUTES FOR OUR API
+// =============================================================================
+const router = express.Router()
+
+router.use(function (req, res, next) {
+  console.log('Something is happening.')
+  next() // make sure we go to the next routes and don't stop here
+})
+
+router.get('/getMinimum/', async function (req, res) {
+  const currencyPair: string = req.query.currencyPair // Param of 'currencyPair' in URI
+  const currentUTCDate: string = formatISO(new Date(), {
+    representation: 'date'
+  }) // Variable for current UTC date using date-fns
+  try {
+    // Array of promises to get the latest documents for the min amount data per plugin
+    const docPromisesArr = swapMinAmts.map(async pluginDb => {
+      return pluginDb.get(currentUTCDate + ':' + currencyPair)
+    })
+    // Capture the result of each promise, including whether it was fulfilled or rejected
+    const docs = await Promise.allSettled(docPromisesArr)
+
+    // Use 'reduce' to go through settled promises and find the lowest minAmount amongst the plugins
+    const minAmountFloat = docs.reduce((currentMinAmount, doc) => {
+      try {
+        const cleanedDoc = asFulfilledPromise(doc) // Clean doc for a fulfilled promise
+        const pluginMinAmount = +cleanedDoc.value.data.minAmount // Grab the plugin's minAmount for the currencyPair
```

Comment:
Why do you have a '+' here?
If it's to cast into a number then this is a hacky way to do it.
You should use 'parseInt' or 'parseFloat'

Reply:
Changed to `parseFloat`.